<a href="https://colab.research.google.com/github/OZZYTA/IntensivoCiclo1/blob/main/Intensivo_For.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Lección 4: For**

Ejemplo 1: for loops

In [ ]:
lista=[0,1,2,3,4,5,6,7]  #lista es una sola variable, con diferentes valores segun su pocision

In [2]:
#Dada una lista de valores, imprimir cada valor de dicha lista
lista=[2,1,5,3,6,8,9,3]

#(inicio, llegada, intervalo)

for i in lista:
  print(i)

2
1
5
3
6
8
9
3


Ejemplo 2: for loops

In [6]:
for i in range(50,100,5): #Punto de partida es limite cerrado, punto de llegada es limite abierto
  print(i)

50
55
60
65
70
75
80
85
90
95


In [ ]:
partida=int(input("Por favor escriba el punto de partida de su recorrido: "))
llegada=int(input("Por favor escriba el punto de llegada de su recorrido: "))
intervalo=int(input("Por favor escriba de cuanto en cuanto desea ir en su recorrido: "))

for i in range(partida, llegada, intervalo):
  print(i)

In [8]:
partida=int(input("Por favor escriba el punto de partida de su recorrido: "))
llegada=int(input("Por favor escriba el punto de llegada de su recorrido: "))
intervalo=int(input("Por favor escriba de cuanto en cuanto desea ir en su recorrido: "))

for i in range(partida, llegada, intervalo):
  print(i, end=" ")

Por favor escriba el punto de partida de su recorrido: 10
Por favor escriba el punto de llegada de su recorrido: 800
Por favor escriba de cuanto en cuanto desea ir en su recorrido: 18
10 28 46 64 82 100 118 136 154 172 190 208 226 244 262 280 298 316 334 352 370 388 406 424 442 460 478 496 514 532 550 568 586 604 622 640 658 676 694 712 730 748 766 784 

Ejemplo 3: for loops

In [10]:
partida=int(input("Por favor escriba el punto de partida de su recorrido: "))
llegada=int(input("Por favor escriba el punto de llegada de su recorrido: "))
intervalo=int(input("Por favor escriba de cuanto en cuanto desea ir en su recorrido: "))

for i in range(partida, llegada+1, intervalo):
  print(i, end=" ")

Por favor escriba el punto de partida de su recorrido: 10
Por favor escriba el punto de llegada de su recorrido: 200
Por favor escriba de cuanto en cuanto desea ir en su recorrido: 5
10 15 20 25 30 35 40 45 50 55 60 65 70 75 80 85 90 95 100 105 110 115 120 125 130 135 140 145 150 155 160 165 170 175 180 185 190 195 200 

Ejemplo 4: arreglos en for loop

In [12]:
semana=["lunes", "martes", "miercoles", "jueves", "viernes", "sabado", "domingo"]

for dia in semana:
  print(dia)

lunes
martes
miercoles
jueves
viernes
sabado
domingo


In [13]:
len(semana)

7

In [14]:
semana[3]

'jueves'

In [16]:
for i in range(0,6+1,1):
  print(semana[i])

lunes
martes
miercoles
jueves
viernes
sabado
domingo


In [29]:
arreglo=["piña", "fresa", "naranja", "limon", "guayaba", "coco", "curuba", "mandarina", "maracuya"]

limite=len(arreglo)
limite

for i in range(0, limite, 1):
  print(arreglo[i])

piña
fresa
naranja
limon
guayaba
coco
curuba
mandarina
maracuya


In [30]:
nombre= "Eduardo"
for i in nombre:
  print(i)

E
d
u
a
r
d
o


In [33]:
numero=str(655.88)
for num in numero:
  print(num)

6
5
5
.
8
8


Example 5: For anidado

In [38]:
comida=["hamburguesa", "pizza", "Pasta"]
bebidas=["Soda", "Jugo", "Agua", "malteada"]

for c in comida:
  for b in bebidas:
    print(c, "con ", b)

hamburguesa con  Soda
hamburguesa con  Jugo
hamburguesa con  Agua
hamburguesa con  malteada
pizza con  Soda
pizza con  Jugo
pizza con  Agua
pizza con  malteada
Pasta con  Soda
Pasta con  Jugo
Pasta con  Agua
Pasta con  malteada


In [37]:
for c in comida:
  for b in range(0,len(bebidas),1):
    print(c, "con ", bebidas[b])

hamburguesa con  Soda
hamburguesa con  Jugo
hamburguesa con  Agua
hamburguesa con  malteada
pizza con  Soda
pizza con  Jugo
pizza con  Agua
pizza con  malteada
Pasta con  Soda
Pasta con  Jugo
Pasta con  Agua
Pasta con  malteada


**Ejercicios de practica/investigación**

1. Desarrollar un programa que le pida al usuario un número entero; luego, el programa debe mostrar su tabla de multiplicar del 1 al 10. Ejemplo: n = 12, entonces: 1 \* 12 = 12, 2 \* 12 = 24...10 \* 12 = 120

2. Desarrolle un programa que solicite al usuario dos números enteros: n y m. El programa debe mostrar la suma de todos los números entre n y m, incluidos n y m.

3. Desarrolle un programa que pida un nombre (una sola palabra) e imprima una a una cada letra del nombre.